### PR TITLE
Try to bring kaniko implementation up-to-date with the latest lifecycle

### PR DESCRIPTION
**Pull request by natalieparellano**
I can confirm that this builds, but not sure if it runs :)

### CODE REVIEW

File: kaniko/code/buildpackconfig/buildPackConfig.go [hunk 1]
```diff
@@ -68,16 +68,7 @@ func NewBuildPackConfig() *BuildPackConfig {
 }
 
 func (b *BuildPackConfig) InitDefaults() {
-
-	logrus.Debug("Check if DOCKER_FILE_NAME env is defined...")
-	b.DockerFileName = util.GetValFromEnVar(DOCKER_FILE_NAME_ENV_NAME)
-	if b.DockerFileName == "" {
-		b.DockerFileName = defaultDockerFileName
-	}
-	logrus.Debugf("DockerfileName is: %s", b.DockerFileName)
```
Comment (natalieparellano) [inline kaniko/code/buildpackconfig/buildPackConfig.go:77]: Removed as this information will be supplied from /layers/config/metadata.toml

File: kaniko/code/buildpackconfig/buildPackConfig.go [hunk 2]
```diff
@@ -92,29 +83,13 @@ func (b *BuildPackConfig) InitDefaults() {
...
-
-	// Convert the CNB ENV vars to Kaniko BuildArgs
-	for k, v := range b.CnbEnvVars {
-		logrus.Debugf("CNB env key: %s, value: %s", k, v)
-		arg := k + "=" + v
-		b.BuildArgs = append(b.BuildArgs, arg)
-	}
-
-	// setup the path to access the Dockerfile within the workspace dir
-	dockerFilePath := b.WorkspaceDir + "/" + b.DockerFileName
```
Comment (natalieparellano) [inline kaniko/code/buildpackconfig/buildPackConfig.go:107]: Same comment as above

File: kaniko/code/buildpackconfig/buildPackConfig.go [hunk 3]
```diff
@@ -5,33 +5,33 @@ import (
...
-	"path"
-	"path/filepath"
-	"strings"
 )
 
 const (
 	homeDir                   = "/"
 	kanikoDir                 = "/kaniko"
-	cacheDir                  = "/cache"
+	cacheDir                  = "/layers/kaniko"
```
Comment (cmoulliard) [inline kaniko/code/buildpackconfig/buildPackConfig.go:31]: Is this path customizable using Lifecycle ? If yes, then we should define an ENV var and externalize it

File: kaniko/code/buildpackconfig/buildPackConfig.go (hunk 1 above)
Comment (cmoulliard) [reply kaniko/code/buildpackconfig/buildPackConfig.go:77]: OK. So we have to review the README.md file to be sure that we do not support/use anymore a dockerfile but instead a TOML file containing the Dockerfile(s)

File: kaniko/code/buildpackconfig/buildPackConfig.go (hunk 2 above)
Comment (cmoulliard) [reply kaniko/code/buildpackconfig/buildPackConfig.go:107]: README should also remove this section explaining how to pass parameters using TOML Dockerfiles and not ENV Vars  ;-)

File: kaniko/code/main.go [hunk 4]
```diff
@@ -115,17 +121,52 @@ func main() {
...
+	logrus.Info("Finding the Dockerfiles to build ...")
+	metadatafileNameToParse := os.Getenv("METADATA_FILE_NAME")
+	logrus.Infof("METADATA TOML FILE: %s", metadatafileNameToParse)
+	if metadatafileNameToParse == "" {
+		metadatafileNameToParse = defaultMetadataFileName
 	}
-	err = reapChildProcesses()
+
+	logrus.Info("Finding base image to provide ...")
+	baseimage := os.Getenv("CNB_BASE_IMAGE")
```
Comment (cmoulliard) [inline kaniko/code/main.go:132]: Is it documented how to pass such ENV var part of the README ? Have you reviewed the Job.yaml definition to align it with your PR - https://github.com/redhat-buildpacks/poc/blob/main/helm/templates/job.yaml#L20-L49 ?

File: kaniko/code/main.go [hunk 5]
```diff
@@ -115,17 +121,52 @@ func main() {
...
+
+		result = append(result, fmt.Sprintf(`base_image=%s`, baseimage))
+		return result
+	}
+
+	for _, d := range meta.Dockerfiles {
+		// Define opts
+		opts := b.Opts
+		opts.DockerfilePath = d.Path
+		opts.BuildArgs = toMultiArg(d.Args)
```
Comment (cmoulliard) [inline kaniko/code/main.go:156]: So finally we will only parse the Build atgs and not the run args here as I see that you changed the model ?

**Review by cmoulliard — COMMENTED** [review]
Great PR. I have some questions and remarks ;-)

File: kaniko/code/buildpackconfig/buildPackConfig.go (hunk 1 above)
Comment (natalieparellano) [reply kaniko/code/buildpackconfig/buildPackConfig.go:77]: (note that the code will still need to be updated further ...forthcoming... to reflect what's in the document, but I think this is closer to where we want to land WRT variable names, etc.)

**Comment by natalieparellano** [comment]
The POC on https://github.com/buildpacks/lifecycle/pull/802 has changed enough that this PR is no longer useful.

**Pull request closed without merging**